nullaway/src/main/java/com/uber/nullaway/handlers/AbstractFieldContractHandler.java

```diff
@@ -201,26 +208,25 @@ protected boolean validateAnnotationSyntax(
   }
 
   /**
-   * Finds a specific field of a class
+   * Finds a specific instance field of a class or its superclasses
    *
    * @param classSymbol A class symbol.
    * @param name Name of the field.
-   * @return The class field with the given name.
+   * @return The field with the given name, or {@code null} if the field cannot be found
    */
-  public static VariableElement getFieldFromClass(Symbol.ClassSymbol classSymbol, String name) {
+  public static @Nullable VariableElement getInstanceFieldOfClass(
+      Symbol.ClassSymbol classSymbol, String name) {
     Preconditions.checkNotNull(classSymbol);
     for (Element member : classSymbol.getEnclosedElements()) {
-      if (member.getKind().isField()) {
+      if (member.getKind().isField() && !member.getModifiers().contains(Modifier.STATIC)) {
         if (member.getSimpleName().toString().equals(name)) {
-          Preconditions.checkArgument(
-              member instanceof VariableElement, "element is not of type: VariableElement");
```

Comment:
Also, why are we removing this precondition check?

Reply:
It's not really a pre-condition, just an invariant.  I'm pretty confident it will never fail.  If it does, we will just get a `ClassCastException` on the next line, and I think that error is just as informative as what this check was doing